staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go

```diff
@@ -618,6 +618,54 @@ type DataImportCronList struct {
 	Items []DataImportCron `json:"items"`
 }
 
+// VolumeImportSource works as a specification to populate PersistentVolumeClaims with data
+// imported from an HTTP/S3/Registry/Blank/ImageIO/VDDK source
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+type VolumeImportSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec VolumeImportSourceSpec `json:"spec"`
+	// +optional
+	Status VolumeImportSourceStatus `json:"status"`
+}
+
+// VolumeImportSourceSpec defines the Spec field for VolumeImportSource
+type VolumeImportSourceSpec struct {
+	ContentType DataVolumeContentType `json:"contentType,omitempty"`
+	// Import sources
```

Comment:
@mhenriks want to know your opinion on this, do you think it's better to have the sources embedded directly in the spec? Or should we add a new field to store them as we do with DataVolumes.

Reply:
Good question!  Yeah I think it makes sense to have this pseudo-union defined in it's own struct